main.tf

```diff
@@ -1 +1,18 @@
+locals {
+  metric_filters = var.create_metric_filter ? { for filter in var.metric_filters : filter.name => filter } : {}
+}
 
+resource "aws_cloudwatch_log_metric_filter" "this" {
+  for_each = local.metric_filters
+
+  name           = each.value["name"]
+  pattern        = each.value["filter_pattern"]
+  log_group_name = each.value["log_group_name"]
+
+  metric_transformation {
```

Comment:
Hmm, so this actually _is_ a `block` declaration, which is internally represented as a list of maps/objects. Does this resource support multiple `metric_transformation` blocks? If so, we can support that by accepting a list of objects for the variable and using a dynamic block. E.g. is something like this valid?

```
resource "aws_cloudwatch_log_metric_filter" "this" {
  name           = ...
  pattern        = ...
  log_group_name = ...

  metric_transformation {
    name          = ...
    namespace     = ...
    value         = ...
    default_value = ...
  }

  metric_transformation {
    name          = ...
    namespace     = ...
    value         = ...
    default_value = ...
  }
}

Reply:
No, it only supports 1 metric_transformation block